### DISCUSSION TITLE
Prettify the module names in LayerChoice

### DISCUSSION THREAD

**Discussion by ultmaster** [microsoft/nni#3781] [Ideas]
Currently, in many cases, LayerChoice induces long names like `net.op_choice.choices.3.conv`. There are two problems:

1. `choices` is basically useless. Users actually want to see `net.op_choice.3.conv`.
2. Use number to label the candidates in LayerChoice is not clear and easy to understand. Users should have the ability to give the modules customized names, like `net.op_choice.sepconv3x3.conv`. This is useful for debugging, visualizing and dumping search space for distributed search.

## How to implement it

For point 1, I think adding the modules directly into `_modules` will work. We can refer to the implementation of `nn.ModuleList`.

For point 2, we can allow the `op_choices` in LayerChoice be more than a list (for example an OrderedDict). Refer to [nn.Sequential](https://pytorch.org/docs/stable/nn.html#torch.nn.Sequential).

As this involves changes of API, let's discuss it first.
